src/ntp/client.c

```diff
@@ -588,52 +589,87 @@ bool ntp_client(const char *server, const bool settime, const bool print)
 static void *ntp_client_thread(void *arg)
 {
 	// Set thread name
-	thread_running[NTP] = true;
-	prctl(PR_SET_NAME, thread_names[NTP], 0, 0, 0);
+	prctl(PR_SET_NAME, thread_names[NTP_CLIENT], 0, 0, 0);
 
 	// Run NTP client
+	bool ntp_server_started = false;
 	bool first_run = true;
 	while(!killed)
 	{
+		// Get time before NTP sync
+		const time_t before = time(NULL);
+
 		// Run NTP client
-		ntp_client(config.ntp.sync.server.v.s, true, false);
+		const bool success = ntp_client(config.ntp.sync.server.v.s, true, false);
+
+		// Get time after NTP sync
+		const time_t after = time(NULL);
 
-		// Load queries from database after first NTP synchronization
-		if(first_run)
+		// If the time was updated by more than one hour, restart FTL to
+		// import recent data. This is relevant when the system time was
+		// set to an incorrect value (e.g., due to a dead CMOS battery
+		// or overall missing RTC) and the time was off.
+		if(first_run && after - before > 3600)
```

Comment:
`after - before > 3600` should we use `abs()` here. Just in case - for any reason -  `before > after`

Reply:
Another question:

Should we use a hard-coded value of 3600 (1 hour) or the value of `ntp.sync.interval.v.ui`?

Reply:
`before > after` means the time has been set back, like from year 2030 to 2024. I don't think this is likely to happen as most updates will either pull the device from Jan 1st, 1970 or from some earlier time stored by `fake-hwclock`. Anyway, it also won't hurt err on the side of caution.

> Should we use a hard-coded value of 3600 (1 hour) or the value of `ntp.sync.interval.v.ui`?

I don't think abusing the interval is a useful compromise here. Users may set the interval to a few seconds ... or even one ... you never know. A sensible constant may be `GCinterval` currently defaulting to 10 minutes.